web/src/echo.js

```diff
@@ -396,7 +396,7 @@ export function process_from_server(messages) {
 
         reify_message_id(local_id, message.id);
 
-        if (message_store.get(message.id).failed_request) {
+        if (message_store.get(message.id)?.failed_request) {
             failed_message_success(message.id);
         }
```

Comment:
Can you explain how this one is possible? I've been confused about it for a long time.

Reply:
I was thinking there is nothing preventing us to receive confirmation of a message being sent successfully with the same `local_id` that we just send a message for and are waiting for confirmation.

Reply:
I'm not sure I understand the theory here. Can you spell out a specific sequence of events?

Reply:
Okay, I looked into a bit more and they indeed seem to be in sync. But if this is happening, there has to be a play with events happening in an unexpected order.